src/praisonai/praisonai/agents_generator.py

```
@@ -579,31 +579,32 @@ def generate_crew_and_kickoff(self):
+                            if isinstance(t, str) and t.strip():
+                                needed_tools.add(t.strip())
+
+                # Resolve only the tools actually referenced in YAML
+                for tool_name in needed_tools:
+                    resolved_tool = self.tool_resolver.resolve(tool_name)
+                    if resolved_tool is None:
+                        self.logger.warning(f"Tool '{tool_name}' not found")
+                        continue
+                    tools_dict[tool_name] = resolved_tool() if inspect.isclass(resolved_tool) else resolved_tool
                             
             except Exception as e:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)